zaptest/observer/observer.go

```diff
@@ -80,6 +80,39 @@ func (o *ObservedLogs) AllUntimed() []LoggedEntry {
 	return ret
 }
 
+// FilterMessage filters entries to those that have the specified message.
+func (o *ObservedLogs) FilterMessage(msg string) *ObservedLogs {
+	return o.filter(func(e LoggedEntry) bool {
+		return e.Message == msg
+	})
+}
+
+// FilterField filters entries to those that have the specified field.
+func (o *ObservedLogs) FilterField(field zapcore.Field) *ObservedLogs {
+	return o.filter(func(e LoggedEntry) bool {
+		for _, ctxField := range e.Context {
+			if ctxField.Key != field.Key {
+				continue
+			}
+			return ctxField == field
```

Comment:
Sure, added a test and supports multiple keys. As soon as any of the fields match with the specified key, the log entry will be returned.